src/main.py

```diff
@@ -1093,7 +1093,7 @@ def _attach_to_primary(self, new_primary, limit):
         return True
 
     def _handle_slots(self):
-        if not self.config.getboolean('global', 'use_replication_slots'):
+        if not self.config.getboolean('global', 'replication_slots_polling'):
```

Comment:
Although you have assigned config value of `replication_slots_polling` as value of `use_replication_slots`, it is not guaranteed that `replication_slots_polling` is True together with `replication_slots_polling`.
I think we should check both options, or forcefully set `replication_slots_polling` to False if `replication_slots_polling` is False.